detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/GuardClauses.kt

```diff
@@ -24,8 +24,8 @@ fun KtNamedFunction.yieldStatementsSkippingGuardClauses(): Sequence<KtExpression
 
 fun KtExpression.isGuardClause(): Boolean {
 
-    fun KtReturnExpression.isIfConditionGuardClause(): Boolean {
-        val ifExpr = this.parent?.parent as? KtIfExpression
+    fun KtReturnExpression.isIfConditionGuardClause(ancestorExpression: KtExpression): Boolean {
```

Comment:
```suggestion
    fun isIfConditionGuardClause(ancestorExpression: KtExpression): Boolean {
```

This can probably be refactored to not be an extension function?

Reply:
I actually like it that way,
and anyway I don't feel comfortable to make that change to someone else's code 🙂

Reply:
+1

Reply:
@cortinico is right here. As you are not using the receiver inside the nested method is it unused.
I would propose an additional change which checks if the return statement is actually the last statement inside the if:

```kotlin
fun KtExpression.isGuardClause(): Boolean {

    fun isIfConditionGuardClause(returnExpr: KtReturnExpression): Boolean {
        val ifExpr = this as? KtIfExpression
        return ifExpr != null &&
            ifExpr.`else` == null &&
            returnExpr === ifExpr.then?.lastBlockStatementOrThis()
    }

    fun KtReturnExpression.isElvisOperatorGuardClause(): Boolean {
        val elvisExpr = this.parent as? KtBinaryExpression
        return elvisExpr != null && elvisExpr.operationToken == KtTokens.ELVIS
    }

    val returnExpr = this.findDescendantOfType<KtReturnExpression>() ?: return false
    return isIfConditionGuardClause(returnExpr) || returnExpr.isElvisOperatorGuardClause()
}

```

Reply:
right
I didn't notice that I no longer use the receiver 😕 
I'll change the code